src/core/nixl_agent.cpp

```diff
@@ -884,6 +885,10 @@ nixlAgent::releaseXferReq(nixlXferReqH *req_hndl) {
 nixl_status_t
 nixlAgent::releasedDlistH (nixlDlistH* dlist_hndl) const {
     NIXL_LOCK_GUARD(data->lock);
+    for (auto it = dlist_hndl->descs.begin(); it != dlist_hndl->descs.end();) {
+        delete it->second;
+        it = dlist_hndl->descs.erase(it);
```

Comment:
can we do it in dlist_hndl destructor?

Reply:
I looked again and found that it was actually cleaned in it destructor, so removed